base/float.jl

```diff
@@ -532,16 +538,17 @@ significand_mask(::Type{Float32}) = 0x007f_ffff
 
 ## Array operations on floating point numbers ##
 
-float{T<:AbstractFloat}(A::AbstractArray{T}) = A
-
-function float{T}(A::AbstractArray{T})
-    if !isleaftype(T)
-        error("`float` not defined on abstractly-typed arrays; please convert to a more specific type")
-    end
-    convert(AbstractArray{typeof(float(zero(T)))}, A)
+# float, broadcast over arrays
+broadcast{T<:AbstractFloat}(::typeof(float), A::AbstractArray{T}) = A
+broadcast(::typeof(float), r::UnitRange) = float(r.start):float(last(r))
+broadcast(::typeof(float), r::StepRange) = float(r.start):float(r.step):float(last(r))
+broadcast(::typeof(float), r::FloatRange) = FloatRange(float(r.start), float(r.step), r.len, float(r.divisor))
+function broadcast(::typeof(float), r::LinSpace)
+    float(r.len) == r.len || error(string(r, ": too long for ", float))
+    LinSpace(float(r.start), float(r.stop), float(r.len), float(r.divisor))
 end
 
-for fn in (:float,:big)
+for fn in (:big,)
```

Comment:
Definitely. `big` is my next target, followed by `real` and `complex`. Then once #16986 hits, `round`, `trunc`, `ceil`, `floor`, and `mod`. Others? Thanks!